File: leetcode/twosum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
const twoSum = function(nums, target) {
    const map = {}
    for (let i = 0; i < nums.length; i++) {
        if (map.hasOwnProperty(nums[i])) return [map[nums[i]], i]
        map[target - nums[i]] = i
    }
}

File: leetcode/twosum.go
package leetcode

import "sort"

func TwoSum(nums []int, target int) []int {
	indexs := make([]int, 2)
	hash := map[int]int{}

	for i := range nums {
		hash[target-nums[i]] = i
	}

	for i := range nums {
		index, ok := hash[nums[i]]
		if ok {
			if i == index {
				continue
			}
			indexs[0] = index
			indexs[1] = i
			sort.Ints(indexs)
			break
		}
		continue
	}

	return indexs
}
